### PR TITLE
add go 1.21 support with dependency updates

### PR DESCRIPTION
**Pull request by MichielBijland**
### ENHANCEMENTS 

Updated dependencies to support go 1.21.x

### CODE REVIEW

File: internal/cloud/auth.go [hunk 1]
```diff
@@ -146,9 +146,13 @@ func getAvailableListener() (net.Listener, error) {
 }
 
 func generatePKCEParams() authhandler.PKCEParams {
+	verifier, err := pkce.NewCodeVerifier(-1)
+	if err != nil {
+		panic("source of randomness unavailable: " + err.Error())
```
Comment (MichielBijland) [inline internal/cloud/auth.go:151]: @demeyerthom don't know if this should be panic or not?

File: internal/cloud/auth.go [hunk 2]
```diff
@@ -145,13 +148,17 @@ func getAvailableListener() (net.Listener, error) {
 		"no free TCP port available in range %d-%d", LocalPortStart, LocalPortEnd)
 }
 
-func generatePKCEParams() authhandler.PKCEParams {
+func generatePKCEParams() (authhandler.PKCEParams, error) {
```
Comment (demeyerthom) [inline internal/cloud/auth.go:151]: :+1:

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       max-parallel: 4
       matrix:
-        go-version: [1.20.x]
+        go-version: [1.20.x, 1.21.x]
```
Comment (demeyerthom) [inline .github/workflows/test.yaml:18]: We are testing for 2 environments here, but only build for 1.21.x. Maybe a better idea to just test for 1.21.x, as that is also what we are distributing our releases with? I don;t think older versions of Go matter very much in that context

**Review by demeyerthom — APPROVED** [review]
LGTM

**Pull request merged**